### PR TITLE
feature: add CLI flag to only sync latest version of charts

### PR DESCRIPTION
**Pull request by tompizmor**
This PR adds a new CLI flag --latest-version-only to only sync the latest version of each chart

### CODE REVIEW

File: pkg/syncer/sync_test.go [hunk 1]
```diff
@@ -263,7 +336,7 @@ func TestSyncPendingChartsChartMuseum(t *testing.T) {
 			tc.targetRepo.GetRepo().Url = tTester.GetURL()
 
 			// Create new syncer
-			s, err := syncer.New(tc.sourceRepo, tc.targetRepo, syncer.WithSkipDependencies(tc.skipDependencies))
+			s, err := syncer.New(tc.sourceRepo, tc.targetRepo, syncer.WithSkipDependencies(tc.skipDependencies), syncer.WithLatestVersionOnly(tc.latestVersionOnly))
```
[the remark this answers is not in the document]
Comment (migmartri) [reply pkg/syncer/sync_test.go:339]: I see that you are following the same pattern we already have in other boolean options `i.e			syncer.WithSkipDependencies(syncSkipDependencies),`

So I guess this is fine then :)

**Review by migmartri — APPROVED** [review]
LGTM!

Do you think we should add this option to the config file?

File: pkg/syncer/index.go [hunk 2]
```diff
@@ -96,45 +99,69 @@ func (s *Syncer) loadCharts(charts ...string) error {
...
-			id := fmt.Sprintf("%s-%s", name, version)
-			klog.V(5).Infof("Details for %q chart: %+v", id, details)
-			if details.PublishedAt.Before(publishingThreshold) {
-				klog.V(5).Infof("Skipping %q chart: Published before %q", id, publishingThreshold.String())
-				continue
-			}
-
-			if ok, err := s.cli.dst.Has(name, version); err != nil {
-				klog.Errorf("unable to explore target repo to check %q chart: %v", id, err)
+			sort.Sort(semver.Collection(vs))
```
[the remark this answers is not in the document]
Comment (tompizmor) [reply pkg/syncer/index.go:111]: Not sure about that as later in the code we are calling to topologicalSortCharts() to take into account the dependencies. We would probably undo the sorting made here

[the remark this answers is not in the document]
Comment (migmartri) [reply pkg/syncer/index.go:111]: Makes sense, thanks!

**Pull request merged**